### PR TITLE
Don't call jac_structure if problem has no constraints

### PR DESCRIPTION
**Pull request by abelsiqueira**
CC @mlubin 

**Untested** because I don't have KNITRO. Follows https://github.com/JuliaOpt/Ipopt.jl/pull/71, fixing for https://github.com/JuliaOpt/MathProgBase.jl/pull/133.

### CODE REVIEW

**Comment by yeesian** [comment]
@mlubin I don't know what to check for (it seems like the unit test is supposed to be added at the MPB level), but this seems acceptable to me. Okay to merge?

**Comment by mlubin** [comment]
If it passes [rosenbrocktest](https://github.com/JuliaOpt/MathProgBase.jl/blob/8339e081a544f12939d765d054cccd6bfb937d17/test/nlp.jl#L301) then you can merge.

File: src/KnitroSolverInterface.jl [hunk 1]
```diff
@@ -76,16 +76,14 @@ function loadproblem!(m::KnitroMathProgModel,
...
-    end
-   
-    Ijac, Jjac = jac_structure(d)
+    init_feat = [:Grad]
+    has_hessian && push!(init_feat, :Hess)
+    numConstr > 0 && push!(init_feat, :Jac)
+    initialize(d, init_feat)
+
+    Ihess, Jhess = has_hessian ? hesslag_structure(d) : (Int[], Int[])
+    Ijac, Jjac = numConstr > 0 && jac_structure(d) : (Int[], Int[])
```
Comment (gragusa) [inline src/KnitroSolverInterface.jl]: `&&` should be `?`

Comment (sylvainmouret) [inline src/KnitroSolverInterface.jl]: I agree

**Comment by sylvainmouret** [comment]
The remaining issue mentioned by gragusa is due to the fact that Knitro asked for objective gradient and jacobian evaluations through a single callback.
Then, the interface calls the two separate callbacks internally.
To fix this, you need to modify the line 254
`kp.eval_jac_g(x,unsafe_wrap(Array,J_,nnzJ))`
by
`if m > 0`
`    kp.eval_jac_g(x,unsafe_wrap(Array,J_,nnzJ))`
`end`

**Comment by huckl3b3rry87** [comment]
@sylvainmouret  line 254 of what script?

**Comment by sylvainmouret** [comment]
Sorry, I forgot to mention it. It's in KNITRO.jl.

**Comment by sylvainmouret** [comment]
This has been implement in #73 so I will close.

**Pull request closed without merging**